source.go

```diff
@@ -77,14 +77,15 @@ func (s *Source) Open(ctx context.Context, pos sdk.Position) error {
 }
 
 func (s *Source) Read(ctx context.Context) (sdk.Record, error) {
-	if s.limiter.Allow() {
-		rec, err := s.getRecord(ctx)
-		if err != nil {
-			return sdk.Record{}, fmt.Errorf("error getting the weather data: %w", err)
-		}
-		return rec, nil
+	err := s.limiter.Wait(ctx)
+	if err != nil {
+		return sdk.Record{}, fmt.Errorf("context was cancelled: %w", err)
```

Comment:
Nitpick: yeah that's the only thing that can happen and the error itself will already contain enough info, no need to wrap it 😉